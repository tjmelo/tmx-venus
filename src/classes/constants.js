export const KEYS = {
  API: 'https://63f239144f17278c9a22d1cd.mockapi.io/api/v1/item',
};

export const ELEM = {
  ADD_ITEM: document.querySelector('.add-item'),
  ALL_ITEMS: document.querySelector('.all-items'),
};
