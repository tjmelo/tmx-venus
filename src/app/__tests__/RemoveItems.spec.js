import toRemoveItems from '../modules/RemoveItems';
import toFetchAPI from '../modules/fetchAPI';

import { KEYS } from '../constants';

jest.mock('../modules/fetchAPI');
jest.mock('../modules/RemoveItems');

describe('Should to remove a determinate items from app', () => {
  it('Should request toFetchAPI for get determinate item', () => {
    const testToFetchAPI = jest.fn();
    testToFetchAPI.mockImplementation((argTest) => toFetchAPI(argTest));

    // do
    testToFetchAPI(KEYS.API);

    // then
    expect(testToFetchAPI).toHaveReturned();
    expect(testToFetchAPI).toHaveBeenCalledTimes(1);
    expect(testToFetchAPI).toHaveBeenCalledWith(KEYS.API);

    // Expects fidelity with scope of function
    expect(testToFetchAPI).toMatchSnapshot();
  });

  it('Should delete an item', () => {
    const testToRemoveItems = jest.fn();
    testToRemoveItems.mockImplementation(() => toRemoveItems());

    const testToFetchAPI = jest.fn();
    testToFetchAPI.mockImplementation((argTest) => toFetchAPI(argTest));

    jest.spyOn(window, 'addEventListener').mockImplementationOnce((event, handler) => {
      const testToDeleteAnItemOnClick = handler(toFetchAPI(`${KEYS.API}/123456789`));
      return testToDeleteAnItemOnClick;
    });

    // do
    testToRemoveItems();
    window.addEventListener('click', (e) => e);

    // then
    expect(testToRemoveItems).toHaveReturned();
    expect(testToRemoveItems).toHaveBeenCalledTimes(1);
    expect(window.addEventListener).toHaveBeenCalledWith('click', expect.any(Function));

    // Expects fidelity with scope of function
    expect(testToRemoveItems).toMatchSnapshot();
  });
});
