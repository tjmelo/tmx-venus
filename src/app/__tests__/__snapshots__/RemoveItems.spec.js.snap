// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should to remove a determinate items from app Should delete an item 1`] = `
[MockFunction] {
  "calls": [
    [],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Should to remove a determinate items from app Should request toFetchAPI for get determinate item 1`] = `
[MockFunction] {
  "calls": [
    [
      "https://63f239144f17278c9a22d1cd.mockapi.io/api/v1/item",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
